# backtest-case: CXX-fused_score_for_moe_aux_loss
# kind: cxx
# base: 42b840051647eef89761a16dfdff87e82bb253ab
# derived: diff(BASE:transformer_engine/common/fused_router/fused_score_for_moe_aux_loss.cu, PRE:transformer_engine/common/fused_router/fused_score_for_moe_aux_loss.cu)
--- a/transformer_engine/common/fused_router/fused_score_for_moe_aux_loss.cu
+++ b/transformer_engine/common/fused_router/fused_score_for_moe_aux_loss.cu
@@ -1,4 +1,6 @@
 /*************************************************************************
+ * This file was modified for portability to AMDGPU
+ * Copyright (c) 2026, Advanced Micro Devices, Inc. All rights reserved.
  * Copyright (c) 2022-2026, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
  *
  * See LICENSE for license information.
@@ -272,7 +274,11 @@
       }
       // Warp reduce the sum
       for (int s = 16; s > 0; s /= 2) {
+#ifdef __HIP_PLATFORM_AMD__
+        local_sum_Output_x_Grad += __shfl_xor(local_sum_Output_x_Grad, s, kThreadsPerWarp);
+#else
         local_sum_Output_x_Grad += __shfl_xor_sync(0xffffffff, local_sum_Output_x_Grad, s);
+#endif
       }
       CompType sum_Output_x_Grad = local_sum_Output_x_Grad;
       // In-place update
